### PR TITLE
Updates to allow --rcm to work again

### PR DESCRIPTION
**Pull request by jwmatthews**
I needed these changes made so I could continue using --rcm to test the release 0.9 built broker. 

I'm intentionally checking in the asb-registry url to make this easier for testing.
Leaving the stage references for those cases where we want to use stage to match what QE is seeing.

### CODE REVIEW

**Comment by jwmatthews** [comment]
Testing a few issues, lets hold off on merging.

**Comment by jwmatthews** [comment]
Waiting for below PR to be available in 3.6 build.
https://github.com/openshift/origin/pull/15486

Related to this BZ:
https://bugzilla.redhat.com/show_bug.cgi?id=1475219

File: ansible/group_vars/all.yml [hunk 1]
```diff
@@ -89,6 +89,9 @@ origin_image_name: docker.io/openshift/origin
 origin_image_tag: v3.6.0-rc.0
 oc_host_config_dir: /var/lib/origin/openshift.local.config
 
+service_catalog_image_name: "origin-service-catalog"
+service_catalog_image_tag: "{{ origin_image_tag }}"
```
Comment (jmrodri) [inline ansible/group_vars/all.yml]: are these even used?

Comment (jwmatthews) [inline ansible/group_vars/all.yml]: They are used just for the syncing of the images.
The name of the image changes from:  origin-service-catalog to ose-service-catalog

Comment (jmrodri) [inline ansible/group_vars/all.yml]: I didn't see the variable used anywhere in the repo which is why I was asking.
```
$ git grep service_catalog_image_name
ansible/group_vars/all.yml:service_catalog_image_name: "origin-service-catalog"
```

Comment (jwmatthews) [inline ansible/group_vars/all.yml]: Thank you, I missed that.  You are right, they aren't used now.  
Updating PR

**Comment by jwmatthews** [comment]
This has been tested with both regular and --rcm.

Looking good.

Ready to merge.

**Comment by dymurray** [comment]
ACK, tested with --rcm flag and this worked for me. I was able to provision without problems.

**Pull request merged**